src/core/PathUtil.cpp

```diff
@@ -97,67 +97,94 @@ namespace lmms::PathUtil
 		}
 	}
 
-	Base baseLookup(const QString & path)
+	Base baseLookup(const QString & input)
 	{
+		if (input.isEmpty()) { return Base::Absolute; };
+
 		for (auto base: relativeBases)
 		{
 			QString prefix = basePrefix(base);
-			if ( path.startsWith(prefix) ) { return base; }
+			if (input.startsWith(prefix)) { return base; }
 		}
 		return Base::Absolute;
 	}
 
 
+	// Enforce forward slashes for cross-platform compatibility.
+	// See issue #8107.
+	QString serializePath(const QString & input)
+	{
+		if (input.isEmpty()) { return input; }
 
+		return QString(input).replace("\\", "/");
+	}
 
-	QString stripPrefix(const QString & path)
+	QString stripPrefix(const QString & input)
 	{
-		return path.mid( basePrefix(baseLookup(path)).length() );
+		if (input.isEmpty()) { return input; }
+
+		QString path = serializePath(input);
+
+		return path.mid(basePrefix(baseLookup(path)).length());
 	}
 
-	QString cleanName(const QString& path)
+	QString cleanName(const QString & input)
 	{
+		if (input.isEmpty()) { return input; }
+
+		QString path = serializePath(input);
+
 		return stripPrefix(QFileInfo(path).completeBaseName());
 	}
 
 
 
-
```

Comment:
You can add this whitespace back.  4 newlines is the old convention, which I'm pretty sure we removed but there's no reason to touch lines you don't have to. https://github.com/LMMS/lmms/wiki/Coding-Conventions